diff --git a/g2o/solvers/CMakeLists.txt b/g2o/solvers/CMakeLists.txt
index 8956453..434ae97 100644
--- a/g2o/solvers/CMakeLists.txt
+++ b/g2o/solvers/CMakeLists.txt
@@ -12,8 +12,8 @@ if(CHOLMOD_FOUND)
 endif(CHOLMOD_FOUND)
 
 # Sparse Module of Eigen is stable starting from 3.1
-if (DEFINED EIGEN3_VERSION)
-  if(${EIGEN3_VERSION} VERSION_GREATER "3.1.0" OR ${EIGEN3_VERSION} VERSION_EQUAL "3.1.0")
+if (DEFINED EIGEN3_VERSION_STRING)
+  if(${EIGEN3_VERSION_STRING} VERSION_GREATER "3.1.0" OR ${EIGEN3_VERSION_STRING} VERSION_EQUAL "3.1.0")
     add_subdirectory(eigen)
   endif()
 endif()
