diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index b59f8e1f98..488f2ce78f 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -1109,6 +1109,7 @@ def _py_wrap_cc_impl(ctx):
   args += [src.path]
   outputs = [ctx.outputs.cc_out, ctx.outputs.py_out]
   ctx.action(
+      use_default_shell_env = True,
       executable=ctx.executable._swig,
       arguments=args,
       inputs=list(inputs),
