diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index adac895a17..2caf8b3f5e 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -1269,6 +1269,7 @@ def _py_wrap_cc_impl(ctx):
     args += [src.path]
     outputs = [ctx.outputs.cc_out, ctx.outputs.py_out]
     ctx.action(
+        use_default_shell_env = True,
         executable = ctx.executable._swig,
         arguments = args,
         inputs = list(inputs),
